client-side/src/stories/header/Icon.jsx

```diff
@@ -1,27 +1,29 @@
-import React from 'react';
-import { Badge, IconButton } from '@mui/material';
-import { MailOutline, Mail } from '@mui/icons-material';
-import './icon.scss';
-
-const MessageIcon = ({ messages = [] }) => {
-  let cntUnreadMessages = messages.filter(message => !message.read).length;
-  const hasUnreadMessages = cntUnreadMessages > 0;
-
-  const handleClick = () => {
-    // Displays a generic component with the list
-  };
-
-  return (
-    <IconButton onClick={handleClick} className="iconButton">
-      {hasUnreadMessages ? (
-        <Badge badgeContent={cntUnreadMessages} color="error" className="badgeContent">
-          <Mail />
-        </Badge>
-      ) : (
-        <MailOutline />
-      )}
-    </IconButton>
-  );
-};
-
-export default MessageIcon;
\ No newline at end of file
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { Badge, IconButton } from '@mui/material';
+import { MailOutline, Mail } from '@mui/icons-material';
+import useWebSocket from '../../webSocket';
+import './icon.scss';
+
+const MessageIcon = () => {
+  const user = useSelector(state => state.user.currentUser);
+  const userId = user._id;
+  const { cntUnreadMessages } = useWebSocket(userId);
+
+  const hasUnreadMessages = cntUnreadMessages > 0;
+
+
+  return (
+    <IconButton className="iconButton">
```

Comment:
Did you mean to remove the management of the open/close the messages component? (I meant only to remove the ; of the onClick, not the whole click event)

Reply:
ok, actually i didnt understand why to remove it...